receptorctl/tests/test_cli.py

```diff
@@ -25,3 +49,38 @@ def test_cmd_status(self, invoke_as_json):
         ) == set(
             json_output.keys()
         ), "The command returned unexpected keys from json output"
+
+    def test_cmd_cancel(self, invoke_as_json, default_receptor_controller_tcp, wait_for_work_finished):
+        # Spawn a long running command
+        node1_controller = default_receptor_controller_tcp
+        status = node1_controller.simple_command("status")
+
+        sleep_for = 5 # in seconds
+
+        work = node1_controller.submit_work("sleep", str(sleep_for), node="node3")
+        result = work.pop("result")
+        unitid = work.pop("unitid")
+
+        assert result == "Job Started"
+        assert wait_for_work_finished(node1_controller, unitid, (sleep_for + 5)), "Workunit timed out and never finished"
+
+        node1_controller.get_work_results(unitid)
```

Comment:
I'll need help with this. I don't know if it's a bug or a feature (no joke), but I'm getting an error here:

```python
assert wait_for_work_finished(node1_controller, unitid, (sleep_for + 5)), "Workunit timed out and never finished"
    
>       node1_controller.get_work_results(unitid)

tests[/test_cli.py:71](): 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <receptorctl.socket_interface.ReceptorControl object at 0x7f40e099c520>
unit_id = '2jUdu2xn', return_socket = False, return_sockfile = True

    def get_work_results(self, unit_id, return_socket=False, return_sockfile=True):
        self.connect()
        self.writestr(f"work results {unit_id}\n")
        text = self.readstr()
        m = re.compile("Streaming results for work unit (.+)").fullmatch(text)
        if not m:
            errmsg = "Failed to get results"
            if str.startswith(text, "ERROR: "):
                errmsg = errmsg + ": " + text[7:]
>           raise RuntimeError(errmsg)
E           RuntimeError: Failed to get results: could not verify signature: signature is empty

receptorctl[/socket_interface.py:255](): RuntimeError
```

Reply:
Was found an error with the TCP connection.
I'll migrate the code to use file sockets only.